### DISCUSSION TITLE
argocd with helm templating

### DISCUSSION THREAD

**Discussion by popdaniel942** [renovatebot/renovate#16788] [Request Help] [answered]
Hello. I have a helm chart that contains some argo application yamls. My usecase is to use renovate bot to update the version of the argo file but I want some values to be taken from the values.yaml file of the helm chart and I want them updated in the same values file. This is because I want to be able to override these values like "targetRevision" when installing the helm chart. And I also have an {{ if }} and {{ end }} so I can choose if I want to install this application or not. Is it possible to use the renovate bot in such a usecase?

{{ if .Values.enabled }}
apiVersion: argoproj.io/v1alpha1
kind: Application
spec:
  source:
    repoURL: {{ .Values.helmRepoUrl }}
    targetRevision: {{ .Values.targetRevision }}
    chart: chart
{{ end }}

**Comment by PhilipAbed** [comment, marked as answer]
i dont think you can depend on one file to update another file unless its a Lockfile. and the support for that is in the documentation.

instead if you are trying to update dependencies in another file then try [RegexManagers](https://docs.renovatebot.com/configuration-options/#regexmanagers)

i hope i understood what you need, cause its not clear to me

**Reply by popdaniel942** [reply]
So I want to update an argocd application. But for it to work I had to hardcode the repoURL and targetRevision. I would like to pass the repoURL and targetRevision from the helm chart's Values.yaml file. But this my argocd application is considered invalid by the renovate bot. I'm trying to find a way to update the argocd application even if it contains helm templating

**Reply by PhilipAbed** [reply]
nope renovate doesn't do that

`values.yaml` we only extract dependencies.
we dont extract variables to be used anywhere else.

**Reply by rarkins** [reply]
@zharinov does this sound similar to what we do for other managers with variables?

**Reply by zharinov** [reply]
Yes, it does

**Reply by PhilipAbed** [reply]
@zharinov @rarkins should we open an issue for this?

**Reply by popdaniel942** [reply]
Actually I managed to make it work with regexManager. I capture the version and chart name from the values.yaml file and then the renovate bot is getting the latest versions from an artifactory. Thank you for giving me the idea of using regexManager

**Reply by pblgomez** [reply]
@popdaniel942 Can you please post the configuration you used for this?

**Reply by popdaniel942** [reply]
In renovatebot.json I have a RegexManager that points to a file, in my case a values.yaml file. Also I point to a registry Url, the artifactory where the helm charts are stored. And then I match some strings in the values file.

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true,
      "matchPaths": [
        "helm/values.yaml"
      ],
      "groupName": "application",
      "matchPackagePatterns": [
        "*"
      ]
    }
  ],
  "regexManagers": [
    {
      "fileMatch": ["helm/values.yaml$"],
      "datasourceTemplate": "helm",
      "registryUrlTemplate": "https://path-to-artifactory",
      "matchStrings": [
        "targetRevision: (?<currentValue>.*?)\n *chartName: (?<depName>.*?)\n"
      ]
    }
  ]
}
```
  ```"targetRevision: (?<currentValue>.*?)\n *chartName: (?<depName>.*?)\n"```   ----> This will search in the values.yaml file for two lines, the first one starting with "targetRevision" and the second one starting with "*chartName" (the * means you can have other characters in the front) and they are separated by a "\n". The "currentValue" is the version which will be updated if a new one is found in the artifactory, and "depName" is the name of the chart to look for in the artifactory.
  You can also combine multiple updates to multiple versions in the same file by using "packageRules"

**Reply by viceice** [reply]
please format code sections

**Reply by pblgomez** [reply]
Would it be possible to have a `values.yaml` with something like

```yaml
---
awsEbsCsiDriver:
  version: 2.10.1
awsNodeTerminationHandler:
  version: 0.19.2
```

and a separate `template/aws-ebs-csi-driver.yaml` with

```yaml
---
apiVersion: argoproj.io/v1alpha1
...
    chart: aws-node-termination-handler
...
    repoURL: https://aws.github.io/eks-charts/
....
    targetRevision: {{.Values.awsEbsCsiDriver.version}}
```

`templates/aws-node-termination-handler.yaml` ... 
`templates/...`

**Reply by pblgomez** [reply]
or the only solution would be to write `chart` `repoURL` and `targetRevision` to values.yaml ?

**Reply by popdaniel942** [reply]
The only way I could make it work was to have the registryUrlTemplate hardcoded. I think you can also use a placeholder for it like ```?<registryUrl>.*?``` but I am not sure. You can make the renovatebot to look in multiple files in ```fileMatch``` part, but I think that all files specified should match the string in the ```matchStrings``` part. So all files need to have in them the chartName, the repoUrl and the version for the matcher to work...Also, having ```targetRevision: {{.Values.awsEbsCsiDriver.version}}``` without using an actual version, I don't think it will work because the renovateBot looks for an actual version, it won't use helm templating to get the value from the Values file. But I'm not an expert in renovatebot so I don't know what it can do.
```
"regexManagers": [
    {
      "fileMatch": ["values.yaml$", "template/aws-ebs-csi-driver.yaml$"],
      "datasourceTemplate": "helm",
      "matchStrings": [
        "chart: (?<depName>.*?)\n *repoUrl: (?<registryUrl>.*?)\n *targetRevision: (?<currentValue>.*?)\n"
      ]
    }
  ]
```

**Reply by bukowa** [reply]
Here is something that works for me
````json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "argocd": {
    "fileMatch": ["\\.yaml$", "\\.yml$"]
  },
  "regexManagers": [
    {
      "matchStrings": [
        "source:\\n.*chart: (?<depName>.*)\\n.*repoURL: (?<registryUrl>.*)\\n.*targetRevision: (?<currentValue>.*)"
      ],
      "fileMatch": ["\\.yaml$", "\\.yml$"],
      "datasourceTemplate": "helm"
    }
  ]
}
````
